overlord/hookstate/hookmgr.go

```diff
@@ -143,13 +153,38 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	// TODO: Actually dispatch the hook.
+	_, err = RunHook(setup.Snap, setup.Revision, setup.Hook)
+	if err != nil {
```

Comment:
I think it would be better if the runCommand error handling is done in the runCommand helper. Something like (if you don't need the output except for the error case which seems to be the case here):
```
func doRunCommand(name string, args...string) error {
  if output, err := exec.Command(name, args...).CombinedOuptut(); err != nil {
    return fmt.Errorf("failed to run command %q %q: %q (%s)", name, args, output, err)
  }
  return nil
}
```
Alternatively https://github.com/snapcore/snapd/blob/master/partition/utils.go#L33 might be something worth putting in osutil if you need the stdout of the command.